### PR TITLE
Don't rely on LazyProxyTrait in LazyServiceEntityRepository

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Makes things simpler and unlocks https://github.com/symfony/symfony/pull/52568

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Could anyone help me with the psalm failure?

**Comment by nicolas-grekas** [comment]
> Error: Property \Doctrine\Bundle\DoctrineBundle\Repository\LazyServiceEntityRepository::$registry does not have native type hint for its value but it should be possible to add it based on @var annotation "ManagerRegistry".

~Wrong hint by phpcs since we should be compatible with PHP 7.4, isn't it?~

nvm, typed props are allowed in 7.4 :facepalm:

File: Repository/LazyServiceEntityRepository.php [hunk 1]
```diff
@@ -18,41 +20,68 @@
...
+    }
 
-        if ($this instanceof LazyObjectInterface) {
-            $initializer($this, '_entityName');
+    /**
+     * @param string|int $name
+     *
+     * @return mixed
+     */
+    public function __get($name)
```
Comment (derrabus) [inline Repository/LazyServiceEntityRepository.php]:
```suggestion
    /** @return mixed */
    public function __get(string $name)
```

I don't understand how `$name` can be an `int`. 🤔

File: Repository/LazyServiceEntityRepository.php [hunk 2]
```diff
@@ -18,41 +20,68 @@
...
-            return;
+        $scope = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['class'] ?? null;
+
+        return (function () use ($name) {
+            return $this->$name;
+        })->bindTo($this, $scope)();
+    }
+
+    /** @param string|int $name */
+    public function __isset($name): bool
```
Comment (derrabus) [inline Repository/LazyServiceEntityRepository.php]:
```suggestion
    public function __isset(string $name): bool
```

**Comment by derrabus** [comment]
> Could anyone help me with the psalm failure?

I have no idea. This ancient Psalm version on this repository has become a burden, tbh.

File: Repository/LazyServiceEntityRepository.php (hunk 2 above)
Comment (nicolas-grekas) [reply Repository/LazyServiceEntityRepository.php]: updated

File: Repository/LazyServiceEntityRepository.php (hunk 1 above)
Comment (nicolas-grekas) [reply Repository/LazyServiceEntityRepository.php]: nvm updated

**Comment by stof** [comment]
@derrabus I suggest that we update the Psalm setup to use Psalm 5 instead. It is probably easier than debugging Psalm 4...

**Comment by ostrolucky** [comment]
Yeah we should update it. I didn't even realize it's so outdated TBH. Psalm 5 supports PHP 7.4, so that shouldn't be an issue either.

**Comment by derrabus** [comment]
> @derrabus I suggest that we update the Psalm setup to use Psalm 5 instead.

I've tried a couple of days ago. If you upgrade Psalm, it spits out seventy-something errors that should be reviewed. Whoever wants to give it a try: be my guest. 🙂

**Review by ostrolucky — APPROVED** [review]

**Comment by derrabus** [comment]
Okay then, screw Psalm for now. This should not block us. Thank you @nicolas-grekas.

**Comment by ostrolucky** [comment]
I'll work on updating to psalm 5 soon

**Pull request merged**